src/main/java/org/magicdgs/readtools/utils/distmap/DistmapEncoder.java

```diff
@@ -0,0 +1,204 @@
+import scala.Tuple2;
+
+/**
+ * Utility class for encode/decode {@link GATKRead} to the Distmap format.
+ *
+ * Distmap (<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0072614">
+ * Pandey & Schlötterer 2013</a>) is a wrapper around different mappers for distributed
+ * computation using Hadoop. The input for this tool is a modified FASTQ format which is writen in
+ * HDFS to safe space and to distribute easily pair-end reads. It is a tab-delimited format with
+ * the following fields:
+ *
+ * 1. Read name with '@' symbol included.
+ * 2. Read sequence.
+ * 3. Read quality.
+ * 4. Second read sequence (if pair-end).
+ * 5. Second read quality (if pair-end).
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public final class DistmapEncoder {
+
+    /** Cannot be instantiated. */
+    private DistmapEncoder() {}
+
+    // Distmap format is tab-delimited
+    private static final String DISTMAP_TOKEN_SEPARATOR = "\t";
+
+    // Distmap format read name is preceded by this character
+    private static final char DISTMAP_READ_NAME_INDICATOR = '@';
+    // read name index in the tokens (common for first and second)
+    private static final int READ_NAME_TOKEN = 0;
+    // first read bases index in the tokens
+    private static final int FIRST_READ_BASES_TOKEN = 1;
+    // first read qualities index in the tokens
+    private static final int FIRST_READ_QUALITY_TOKEN = 2;
+    // second read bases index in the token
+    private static final int SECOND_READ_BASES_TOKEN = 3;
+    // second read qualities indes in the tokens
+    private static final int SECOND_READ_QUALITY_TOKEN = 4;
+    // maximum number of tokens in the index
+    private static final int MAXIMUM_NUMBER_OF_TOKENS = 5;
+
+    /**
+     * Encodes a single-end read into a Distmap string.
+     *
+     * @param read the read to encode.
+     */
+    public static final String encode(final GATKRead read) {
+        Utils.nonNull(read, "null read");
+        return String.format("%s%s\t%s\t%s",
+                DISTMAP_READ_NAME_INDICATOR,
+                RTReadUtils.getReadNameWithIlluminaBarcode(read),
+                read.getBasesString(),
+                ReadUtils.getBaseQualityString(read)
+        );
+    }
+
+    /**
+     * Encodes a pair-end read into a Distmap string.
+     *
+     * Note: barcode information is not maintained yet.
+     *
+     * @param pair tuple with the first and the second reads in the pair.
+     */
+    public static final String encode(final Tuple2<GATKRead, GATKRead> pair) {
+        Utils.nonNull(pair, "null pair-end read");
+        Utils.nonNull(pair._1, "null first read");
+        Utils.nonNull(pair._2, "null second read");
+        Utils.validateArg(pair._1.getName().equals(pair._2.getName()),
+                () -> "not equal names for pairs: " + pair._1 + " vs. " + pair._2);
+        // TODO: check if barcodes are equal? - it is only using the first pair for that...
```

Comment:
Requires to open an issue to track this, just in case it becomes important. We are assuming that all across the toolkit....